compression/compress.h

```diff
@@ -79,6 +79,9 @@ class CompressedArray {
   MatT* data() { return data_.data(); }
   const MatT* data() const { return data_.data(); }
 
+  MatT& operator[](size_t pos) { return data_[pos]; }
```

Comment:
Are we OK with this not working for MatT != bfloat16_t and float?  Maybe safest to remove, so that users are encouraged to go through compress-inl functions?